### ISSUE TITLE
Update Transmission to 2.94

### ISSUE THREAD

**Issue by nexusmaniac** [haugene/docker-transmission-openvpn#510]
It's not an 'issue' per se. But I'd love to see this fantastic container get updated with the latest Transmission commits! 😄 

Latest release was a a couple of weeks ago:
https://github.com/transmission/transmission/releases/tag/2.94

**Comment by coolboiime** [comment]
The script installs the latest Transmission. I rebooted my Docker container yesterday, and I have 2.93 currently.

**Comment by nexusmaniac** [comment]
Nope, I've just restarted my docker container and I'm definitely still running 2.92 😛 

I'm running unRAID just for reference.

**Comment by coolboiime** [comment]
I deleted my docker container and re-ran the script.

**Comment by nexusmaniac** [comment]
I've just deleted my docker image and re-pulled the container. Essentially a 'fresh install' of it. (dev-alpine) and still on `Transmission 2.92 (14714)`

Please point me in the right direction if I'm going mad 😉

**Comment by coolboiime** [comment]
Huh! That us extremely weird. I don’t find it in the repos.

**Comment by nexusmaniac** [comment]
Find what in which repos? Apologies, I didn't follow that last message 😄 

My setup simple pulls the container from DockerHub, https://hub.docker.com/r/haugene/transmission-openvpn/tags/

And I run the `dev-alpine` tag 👍 

Are you referring to the docker-compose script?

**Comment by coolboiime** [comment]
I was checking the Dockerfile for the repo, and I assumed that I could search for the latest version, but it ended up returning no results.

My assumption is that the repo doesn’t have a 2.94 version, cause it’s doing

`apk transmission-daemon`

I don’t work with alpine linux much, but this is what I found😁

**Comment by nexusmaniac** [comment]
I hadn't thought about that 😄

You are correct! I just jumped inside the container to check which version it was running / what was available:

```
/ # apk version transmission-daemon
WARNING: This apk-tools is OLD! Some packages might not function properly.
Installed:                                Available:
transmission-daemon-2.92-r8             = 2.92-r8
/ #
```

**Comment by nexusmaniac** [comment]
And yet:

https://pkgs.alpinelinux.org/package/edge/main/x86/transmission-daemon

Seems that the `Main` alpine repo has version 2.94 🤔 ... 

```
/ # cat /etc/apk/repositories
http://dl-cdn.alpinelinux.org/alpine/v3.7/main
http://dl-cdn.alpinelinux.org/alpine/v3.7/community
@testing http://dl-cdn.alpinelinux.org/alpine/edge/testing
/ #
```

Can't see why it's not seeing / fetching the latest even when trying to do so manually 😛

**Comment by coolboiime** [comment]
Do they update apk-tools at all😆

**Comment by coolboiime** [comment]
I believe it’s because they’re using the testing repo, the main repo has the most up to date packages.

**Comment by nexusmaniac** [comment]
Damn, this is where my knowledge of Apline / APK expires ;)

I have no idea how to force `apk` to download from the edge branch.

What I discovered is that Alpine edge (master?) has Transmission 2.94 whereas Alpine 3.7 has 2.92-r8 😖 

Just need to find a way to force-pull from the 'edge' branch 👍

**Comment by coolboiime** [comment]
Let me know how it goes!

**Comment by nexusmaniac** [comment]
🎉 🎉 🎉 

```
Transmission 2.94 (d8e60ee44f)

A fast and easy BitTorrent client

Copyright (c) The Transmission Project
```

Excellent!! 👍  That was easier than anticipated 😉 

------

Plenty of help from: https://wiki.alpinelinux.org/wiki/Alpine_Linux_package_management

I copied their `/etc/apk/repositories` config:
```
http://nl.alpinelinux.org/alpine/v3.7/main
http://nl.alpinelinux.org/alpine/v3.7/community
@edge http://nl.alpinelinux.org/alpine/edge/main
@edgecommunity http://nl.alpinelinux.org/alpine/edge/community
@testing http://nl.alpinelinux.org/alpine/edge/testing
```

And ran `apk update` & `apk upgrade`

I then ran `apk del transmission-daemon` & `apk add transmission-daemon@edge` restarted the container and success 💯 

All up and running on the latest version of Transmission-BT 😄

**Comment by coolboiime** [comment]
Nice work!

**Comment by nexusmaniac** [comment]
Cheers for the help, I wouldn't have thought about doing any of this without your input! 👍 

Guess I should close this ticket. Presumably once Alpine gets update / Transmission gets added to the Alpine 3.7 repo/branch thing it'll just auto update as you say 😎

**Comment by coolboiime** [comment]
Sounds good😎😎

**Comment by LarsaSolidor** [comment]
Sorry to open a closed issue, but I didn't want to create a whole new issue to ask. 

What are the steps to get tranmission 2.94 working on the arm image, for use on the raspberry pi? I'm getting 2.92.

**Comment by ernstp** [comment]
Not sure why this was closed but now it looks like it's actually closed! #1026
